tournaments/index.js

```diff
@@ -1152,33 +1152,33 @@ let commands = {
 			}
 			tournament.customRules = [];
 			tournament.teambuilderFormat = tournament.originalFormat;
-			this.room.addRaw("<b>The tournament's custom rules were cleared.</b>");
-			this.privateModAction("(" + user.name + " cleared the tournament's custom rules.)");
+			this.room.addRaw(`<b>The tournament's custom rules were cleared.</b>`);
+			this.privateModAction(`(${user.name} cleared the tournament's custom rules.)`);
 			this.modlog('TOUR CLEARRULES');
 		},
 		name: 'setname',
 		customname: 'setname',
 		setname: function (tournament, user, params, cmd) {
 			if (params.length < 1) {
-				return this.sendReply("Usage: " + cmd + " <comma-separated arguments>");
+				return this.sendReply(`Usage: ${cmd} <comma-separated arguments>`);
 			}
 			let name = this.canTalk(params[0].trim());
 			if (!name) return;
 			name = Chat.escapeHTML(name);
-			if (name.length > MAX_CUSTOM_NAME_LENGTH) return this.errorReply("The tournament's name cannot exceed " + MAX_CUSTOM_NAME_LENGTH + " characters.");
+			if (name.length > MAX_CUSTOM_NAME_LENGTH) return this.errorReply(`The tournament's name cannot exceed ${MAX_CUSTOM_NAME_LENGTH} characters.`);
 			if (name.includes('|')) return this.errorReply("The tournament's name cannot include the | symbol.");
 			tournament.format = name;
-			this.room.send('|tournament|update|' + JSON.stringify({format: tournament.format}));
-			this.privateModAction("(" + user.name + " set the tournament's name to " + tournament.format + ".)");
+			this.room.send(`|tournament|update|${JSON.stringify({format: tournament.format})}`);
```

Comment:
IIRC, last time it was agreed that no function calls would be concatenated inside template strings.

Reply:
With the exception of, like, `Chat.plural` I guess

Reply:
So, what do I have to change here?

Reply:
Make a temporary variable for `JSON.stringify({format: tournament.format})`

Reply:
Seriously? It’s ridiculous to make a temporary variable for a single, simple function call just because you’re using ` symbols instead of ' symbols, especially if that function is just `JSON.stringify`. That’s just asking for naming conflicts and typos. I can understand doing that for particularly long or complicated statements, but both this and the original version of that line are exactly the same length, and are each short enough to fit within the width of my phone’s screen in landscape mode. If you’re going to enforce “only very simple function calls inside strings,” why not be consistent and apply that to _all_ strings?

Reply:
On second thought, it’s also seems a little silly to put such a simple object literal through  a call to `JSON.stringify`. This could just as easily be `|tournament|update|{"format":"${tournament.format}"}` or even just `|tournament|update|{"format":"${name}"}`

Reply:
For readability, `JSON.stringify` is nice. I wouldn't consider the tradeoff of manually constructing an object literal to be worth it.

I guess I can allow `JSON.stringify`, BUT there are still two other things wrong with this:

1. You're manually constructing an object to pass to `JSON.stringify`, which is not something I want done inside a template string.

2. It's over 80 columns wide when tab width is set to 3 spaces, which is my limit.

I think I want an intermediate variable for the object. I find intermediate variables generally more readable than stuffing too much into one line, anyway.

Reply:
There are four things going on in this line:

- protocol object `{format: tournament.format}` is assembled
- `JSON.stringify` is called
- the buffer to be sent over the wire is constructed
- `this.room.send` is called

This is a lot going on in a single line. To be fair, the last three are related, and I'm fine with those being on one line. But all four does seem like a bit much, to me. When I write this kind of code, it would be:

```js
const update = {
  format: tournament.format,
};
this.room.send(`|tournament|update|${JSON.stringify(update)}`);
```

Example:
https://github.com/Zarel/Pokemon-Showdown/blob/8b1f9b69f411f6ed40cef3b4796ff46e7b06354b/sim/battle.js#L3303-L3314